app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart

```diff
@@ -162,53 +159,41 @@ class CheckForWaitingPullRequests extends ApiRequestHandler<Body> {
     }
 
-    final Map<String, dynamic> label =
-        repository['labels']['nodes'].single as Map<String, dynamic>;
-    if (label == null || label.isEmpty) {
+    if (data.repository.labels.nodes.isEmpty) {
       throw StateError(
           'Query did not find information about the waitingForTreeToGoGreen label.');
     }
-    final String labelId = label['id'] as String;
+    final String labelId = data.repository.labels.nodes[0].id;
     final List<_AutoMergeQueryResult> list = <_AutoMergeQueryResult>[];
-    final Iterable<Map<String, dynamic>> pullRequests =
-        (label['pullRequests']['nodes'] as List<dynamic>)
-            .cast<Map<String, dynamic>>();
-    for (Map<String, dynamic> pullRequest in pullRequests) {
-      final Map<String, dynamic> commit = pullRequest['commits']['nodes']
-          .single['commit'] as Map<String, dynamic>;
+    final List<$LabeledPullRequcodeestsWithReviews$repository$labels$nodes$pullRequests$nodes> pullRequests = data.repository.labels.nodes[0].pullRequests.nodes;
+    for ($LabeledPullRequcodeestsWithReviews$repository$labels$nodes$pullRequests$nodes pullRequest in pullRequests) {
+      final $LabeledPullRequcodeestsWithReviews$repository$labels$nodes$pullRequests$nodes$commits$nodes$commit commit = pullRequest.commits.nodes[0].commit;
       // Skip commits that are less than an hour old.
       // Use the committedDate if pushedDate is null (commitedDate cannot be null).
-      final DateTime utcDate = DateTime.parse(commit['pushedDate'] as String ??
-              commit['committedDate'] as String)
-          .toUtc();
+      final DateTime utcDate = DateTime.parse((commit.pushedDate ?? commit.committedDate).value).toUtc();
       if (utcDate
           .add(const Duration(hours: 1))
           .isAfter(DateTime.now().toUtc())) {
         continue;
       }
-      final String author = pullRequest['author']['login'] as String;
-      final String id = pullRequest['id'] as String;
-      final int number = pullRequest['number'] as int;
+      final String author = pullRequest.author.login;
+      final String id = pullRequest.id;
+      final int number = pullRequest.number;
 
       final Set<String> changeRequestAuthors = <String>{};
       final bool hasApproval = config.rollerAccounts.contains(author) ||
           _checkApproval(
-            (pullRequest['reviews']['nodes'] as List<dynamic>)
-                .cast<Map<String, dynamic>>(),
+            pullRequest.reviews.nodes,
             changeRequestAuthors,
           );
 
-      final String sha = commit['oid'] as String;
-      final List<Map<String, dynamic>> statuses =
-          (commit['status']['contexts'] as List<dynamic>)
-              .cast<Map<String, dynamic>>();
+      final String sha = commit.oid.value;
+      final List<$LabeledPullRequcodeestsWithReviews$repository$labels$nodes$pullRequests$nodes$commits$nodes$commit$status$contexts> statuses = commit.status.contexts;
```

Comment:
As for such long type references, not sure if is there any way to have short replacement names? like defining type mapping in a central location, and then refer the short name here.